src/frontend/utils/chat/chat.ts

```diff
@@ -23,3 +24,89 @@ export const getNameFromJID = (jid: string): string => {
 export const isAnonymous = (name: string) => {
   return name.toLowerCase().startsWith(`${ANONYMOUS_ID_PREFIX}-`);
 };
+
+interface Segment {
+  startIndex: number;
+  stopIndex: number;
+}
+
+export const locateUrlSegments = (textToProcess: string): Segment[] => {
+  const resultArray: Segment[] = [];
+  let lastIndexProcessed = 0;
+
+  const arrayOfMatches = linkify.find(textToProcess, 'url');
+
+  // Regex returns also the space at the end of an url
+  const trimmedArray = arrayOfMatches.map((match) => match.value.trimEnd());
+  trimmedArray.forEach((url) => {
+    const startIndex = textToProcess.indexOf(url, lastIndexProcessed);
+    lastIndexProcessed = startIndex + url.length - 2;
+    resultArray.push({ startIndex, stopIndex: startIndex + url.length - 1 });
+  });
+
+  return resultArray;
+};
+
+interface DifferentiateText {
+  type: 'text' | 'segment';
+  content: string;
+}
+
+export const differentiateTextAndSegment = (
+  textToProcess: string,
+  segmentsArray: Segment[],
+): DifferentiateText[] => {
+  const differentiatedTextArray: DifferentiateText[] = [];
+
+  const firstSegment = segmentsArray[0];
+  const lastSegment = segmentsArray[segmentsArray.length - 1];
+
+  if (firstSegment.startIndex !== 0) {
+    differentiatedTextArray.push({
+      type: 'text',
+      content: textToProcess.slice(0, segmentsArray[0].startIndex),
+    });
+  }
+
+  for (let i = 0; i < segmentsArray.length; i++) {
+    const actualSegment = segmentsArray[i];
+    const nextSegment =
+      i + 1 < segmentsArray.length ? segmentsArray[i + 1] : null;
+    differentiatedTextArray.push({
+      type: 'segment',
+      content: textToProcess.slice(
+        actualSegment.startIndex,
+        actualSegment.stopIndex + 1 !== textToProcess.length
+          ? actualSegment.stopIndex + 1
+          : undefined,
+      ),
+    });
+
+    if (nextSegment) {
+      differentiatedTextArray.push({
+        type: 'text',
+        content: textToProcess.slice(
+          actualSegment.stopIndex + 1,
+          nextSegment.startIndex,
+        ),
+      });
+    }
+  }
+
+  if (lastSegment.stopIndex + 1 !== textToProcess.length) {
+    differentiatedTextArray.push({
+      type: 'text',
+      content: textToProcess.slice(lastSegment.stopIndex + 1),
+    });
+  }
+  return differentiatedTextArray;
+};
+
+export const formatUrl = (urlToFormat: string): string => {
+  if (
+    !(urlToFormat.startsWith('http://') || urlToFormat.startsWith('https://'))
```

Comment:
```suggestion
    !(urlToFormat.startsWith('http://') || !urlToFormat.startsWith('https://'))
```

Reply:
Is it still relevant to do that with linkify ?

Reply:
And can it starts with something else ? I mean if it starts by `ftp://` you will have a link stating like this `//ftp://` and it's invalid.

Reply:
Yes because we use linkifyjs only to identify urls in text, not to generate correct url from incomplete ones.

Yes I changed urlFormat to also accept url (because it's not something you can configure on linkifyjs)